File: 8_sumarray.cpp
#include <iostream>

// Recursive function to sum array elements
int sumArray(const int arr[], int size) {
    // Base case: if size is 0, sum is 0
    if (size == 0) {
        return 0;
    }
    // Recursive case: current element + sum of remaining elements
    return arr[size - 1] + sumArray(arr, size - 1);
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int size = sizeof(arr) / sizeof(arr[0]);
    std::cout << "Sum of array elements: " << sumArray(arr, size) << std::endl;
    return 0;
}

File: 8_sumarray.py
def sum_array(arr):
    # Если длина = 0, вернуть 0
    if not arr:
        return 0
    # рекурсия - значение первого элемента + значения из списка без первого элемента
    return arr[0] + sum_array(arr[1:])
    
arr = [1,2,34,5,4,33,2,0]
print(sum_array(arr)) # 81
